### DISCUSSION TITLE
Subscription returning error

### DISCUSSION THREAD

**Discussion by Razorholt** [ctwillie/expo-server-sdk-php#6] [Q&A] [answered]
Hello -

I get this error after sending the message:
`Array ( [0] => Array ( [status] => error [message] => "ExponentPushToken[*******************]" is not a registered push notification recipient [details] => Array ( [error] => DeviceNotRegistered ) ) )`

I check the response from the subscription and it returns the value '1". Also, the expo.json file shows this:
`{"news-letter":["ExponentPushToken[*******************]"]}`

Here is my code:

```
require '/home/project/vendor/autoload.php';

use ExpoSDK\ExpoMessage;
use ExpoSDK\Expo;

$expo = Expo::driver('file');

$message = (new ExpoMessage())
    ->setTitle('Message Title')
    ->setSubtitle('Message sub title')
    ->setBody('The notification message body')
    ->setChannelId('default')
	->setData(['foo' => 'bar'])
    ->setBadge(0)
    ->playSound();

$recipients = [
    'ExponentPushToken[*******************]',
];

$channel = 'news-letter';
$expo->subscribe($channel, $recipients);

$response = $expo->send($message)->toChannel($channel)->push();
$data = $response->getData();
print_r($data);
```
Thanks!

**Comment by ctwillie** [comment, marked as answer]
@Razorholt If you are using this expo push token, ```ExponentPushToken[*******************]```, then that is the exact error you would get, as it is not an actual expo push token. You must use a token generated by expo.

Being this is an expo specific question and not really an issue related to this package, I would again suggest you spend some time reading through the expo docs on push notifications. The docs explain how to generate tokens, send notifications and retrieve push receipts. Understanding their docs will greatly help with working with this package.

**Reply by Razorholt** [reply]
No, I didn't use the token I mentioned - I didn't want people to get the real one. lol. However, it was my mistake because I just realized that I used a made up one... Sorry about that :(

You're the best. Keep up the good job, man!

**Reply by ctwillie** [reply]
No, problem. Glad you figured it out. Thanks!
